deployment/base/hg-deployment.yaml

```diff
@@ -131,3 +131,33 @@ spec:
           items:
           - key: hgweb.hgrc
             path: hgweb.hgrc
+
+      initContainers:
+      - name: init-repo-structure
+        securityContext:
+          runAsUser: 33
+          runAsGroup: 33 # www-data
+        image: busybox:1.36.1
+        command:
+          - 'sh'
+          - '-c'
+          - |
+            cd /repos
+            for folder in /repos/[a-z0-9]?*/; do
+              project=$(basename "$folder")
+              if [[ -L "$project" ]]; then
+                echo "Skipping symlinked folder: $project"
+                continue
+              fi
+              first_letter=$(echo "$project" | head -c1)
+
+              mkdir -p /repos/"$first_letter"/
```

Comment:
This might be unnecessary if we move the two `mkdir -p /repos/a /repos/b ...` lines from the bottom of the script to the top of the script. But we might as well keep it for safety's sake, I don't think it costs a lot of time to run a mkdir -p that makes no changes to the filesystem.